tensorcircuit/quantum.py

```diff
@@ -1153,6 +1153,91 @@ def generate_local_hamiltonian(
 
 
 # TODO(@Charlespkuer): Add more conversion functions for other packages
+def extract_tensors_from_qop(qop: QuOperator) -> Tuple[List[Node], bool, int]:
+    """
+    Extract and sort tensors from QuOperator for conversion to other tensor network formats.
+
+    :param qop: Input QuOperator
+    :return: Tuple of (sorted_nodes, is_mps, nwires)
+    """
+    is_mps = len(qop.in_edges) == 0
+    nwires = len(qop.out_edges) if is_mps else len(qop.in_edges)
+
+    # Collect all nodes from edges
+    all_nodes_from_edges = set()
+    for edge in qop.in_edges + qop.out_edges + qop.ignore_edges:
+        if edge.node1 is not None:
+            all_nodes_from_edges.add(edge.node1)
+        if edge.node2 is not None:
+            all_nodes_from_edges.add(edge.node2)
+
+    nodes_for_sorting = (
+        list(all_nodes_from_edges) if not qop.ref_nodes else qop.ref_nodes
+    )
+
+    # Find endpoint nodes
+    endpoint_nodes = set()
+    physical_edges = set(qop.out_edges) if is_mps else set(qop.in_edges + qop.out_edges)
+
+    if is_mps:
+        rank_2_nodes = {node for node in nodes_for_sorting if len(node.edges) == 2}
+        if len(rank_2_nodes) == 2:
+            endpoint_nodes = rank_2_nodes
+
+    if not endpoint_nodes:
+        endpoint_nodes = {edge.node1 for edge in qop.ignore_edges if edge.node1}
+
+    if not endpoint_nodes and len(nodes_for_sorting) > 1:
```

Comment:
what is the case here, the min bond should be counted as sum of the bond dimension instead of sum of 1?